esmvalcore/_recipe/recipe.py

```diff
@@ -593,23 +567,28 @@ def _get_input_files(variable, config_user):
     # Set up downloading from ESGF if requested.
     if (not config_user['offline']
             and variable['project'] in esgf.facets.FACETS):
-        try:
-            check.data_availability(
-                input_files,
-                variable,
-                globs,
-                log=False,
-            )
-        except RecipeError:
+        search_esgf = config_user['always_search_esgf']
+        if not search_esgf:
             # Only look on ESGF if files are not available locally.
+            try:
+                check.data_availability(
+                    input_files,
+                    variable,
+                    globs,
+                    log=False,
+                )
+            except RecipeError:
+                search_esgf = True
```

Comment:
this is too much under the hood, I think a debug message would be appropriate here, what say you?

Reply:
aha! Happy to see one, then we skip this